src/lib/output/plugins/HierarchyPlugin.ts

```diff
@@ -0,0 +1,59 @@
+import * as Path from "path";
+import { Component, RendererComponent } from "../components";
+import { RendererEvent } from "../events";
+import { writeFile } from "../../utils";
+import { DefaultTheme } from "../themes/default/DefaultTheme";
+import { gzip } from "zlib";
+import { promisify } from "util";
+import { type Reflection } from "../../models";
+import { UrlMapping } from "../models/UrlMapping";
+
+const gzipP = promisify(gzip);
+
+@Component({ name: "hierarchy" })
+export class HierarchyPlugin extends RendererComponent {
+    override initialize() {
+        this.owner.on(RendererEvent.BEGIN, this.onRendererBegin.bind(this));
+    }
+
+    private onRendererBegin(_event: RendererEvent) {
+        if (!(this.owner.theme instanceof DefaultTheme)) {
+            return;
+        }
+
+        this.owner.preRenderAsyncJobs.push((event) =>
+            this.buildHierarchy(event),
+        );
+    }
+
+    private async buildHierarchy(event: RendererEvent) {
+        const theme = this.owner.theme as DefaultTheme;
+
+        const [_, pageEvent] = event.createPageEvent<Reflection>(
```

Comment:
Wow, I did that at the start and forgot to go back to do it properly. Egg on my face. I'll get started on fixing this soon, but it might take me a bit to get to it.

Reply:
Wait, I remember why I did this initially, although I do still want to find a better way. `getHierarchy` relies on the `type` partial to render the `html` property for each element, which takes a `DefaultThemeRenderContext` argument. The only way I found to create a full `DefaultThemeRenderContext` instance was to fake a `PageEvent`. I'll look for an alternative, but do you know of a better way to handle this?

As for `getHierarchy` being on the context, very true, that should be easy enough to fix.

Reply:
Ah, ew.. that's very unfortunate. I'm going to say don't worry about it for now.  I'll look for a better way to do this when considering breaking changes for 0.27.

Reply:
Sounds good to me, I took another look around and didn't find a good alternative that wouldn't be a breaking change.